libraries/botbuilder-ai-orchestrator/src/orchestratorAdaptiveRecognizer.ts

```diff
@@ -110,228 +90,196 @@ export class OrchestratorAdaptiveRecognizer extends Recognizer implements Orches
-        if (detectAmbiguity) {
-            const topScoringIntent = result[0].score;
-            const scoreDelta = topScoringIntent - disambiguationScoreThreshold;
-            const ambiguousIntents = result.filter((x) => x.score >= scoreDelta);
-            if (ambiguousIntents.length > 1) {
-                recognizerResult.intents = {};
-                recognizerResult.intents[this.chooseIntent] = { score: 1.0 };
-                recognizerResult[this.candidatesCollection] = [];
-                ambiguousIntents.forEach((item) => {
-                    const itemRecoResult = {
-                        text: text,
-                        intents: {},
-                        entities: {},
-                        score: item.score,
-                    };
-                    itemRecoResult.intents[item.label.name] = {
-                        score: item.score,
-                    };
-                    itemRecoResult.entities = recognizerResult.entities;
-                    recognizerResult[this.candidatesCollection].push({
-                        intent: item.label.name,
-                        score: item.score,
-                        closestText: item.closest_text,
-                        result: itemRecoResult,
-                    });
-                });
+            // if top scoring intent is less than threshold, return None
+            if (topScore < this.unknownIntentFilterScore) {
+                recognizerResult.intents.None = { score: 1.0 };
+            } else {
+                // add top score
+                recognizerResult.intents[`${topScoringIntent}`] ??= {
+                    score: topScore,
+                };
+
+                // disambiguate
+                if (detectAmbiguity) {
+                    const disambiguationScoreThreshold = this.disambiguationScoreThreshold.getValue(dc.state);
+                    const classifyingScore = topScore - disambiguationScoreThreshold;
+                    const ambiguousResults = results.filter(
+                        (item: { score: number }) => item.score >= classifyingScore
+                    );
+                    if (ambiguousResults.length > 1) {
+                        const recognizerResults: Record<string, RecognizerResult> = ambiguousResults
+                            .map(
+                                (result): RecognizerResult => ({
+                                    text,
+                                    alteredText: result.closest_text,
+                                    entities: recognizerResult.entities,
+                                    intents: { [result.label.name]: { score: result.score } },
+                                })
+                            )
+                            .reduce((results: Record<string, RecognizerResult>, result: RecognizerResult) => {
+                                const guid = generate_guid();
+                                results[`${guid}`] = result;
```

Comment:
Same question - is the interpolation necessary? `results[guid]` should be fine right?